doc/contributing/fpga/debugging_with_ila.md

```diff
@@ -0,0 +1,280 @@
+
+Once Vivado has successfully generated a bitstream, locate its directory with `dirname $(./bazelisk.sh outquery-all //hw/bitstream:rom --define bitstream=vivado)` (it will usually be `bazel-out/k8-fastbuild/bin/hw/bitstream/vivado`).
+Append `/build.fpga_cw310/synth-vivado` to that path.
+In the resulting directory, you should find `otp.mmi`, `rom.mmi`, and `lowrisc_systems_chip_earlgrey_cw310_0.1.bit`.
+Copy those files into the `hw/bitstream` directory and rename them as follows:
+- `otp.mmi` -> `local_otp.mmi`
+- `rom.mmi` -> `local_rom.mmi`
+- `lowrisc_systems_chip_earlgrey_cw310_0.1.bit` -> `local_chip_earlgrey_cw310.bit`
+
+Now invoke `bazel build //hw/bitstream:rom_otp_test_unlocked0 --define bitstream=local` to splice the desired ROM and OTP image into the bitstream with an ILA.
+The resulting bitstream will be located with `./bazelisk.sh outquery-all //hw/bitstream:rom_otp_test_unlocked0 --define bitstream=local`.
+
+
+## Programming the FPGA, controlling the ILA, and running a test
+
+Start the Vivado GUI on the workstation to which the Platform Cable USB II is connected and click *Open Hardware Manager*.
+Open a hardware target by clicking *Open target* -> *Auto Connect*.
+If that does not work, click *Open New Target...* (instead of *Auto Connect*) and *Next*, then select *Local server* and click *Next*.
+If there are no hardware targets in this window, check that Platform Cable USB II is properly connected, its drivers installed on the workstation, and the user under which you run Vivado has access to USB devices.
+Otherwise, select the only targets and click *Next* and *Finish* to connect to the FPGA.
+
+Click *Program device*.
+As *Bitstream file*, select the file identified at the end of the previous section.
+As *Debug probes file*, navigate to the `synth-vivado` directory identified in the previous section, then go to `lowrisc_systems_chip_earlgrey_cw310_0.1.runs/impl_1` and select `debug_nets.ltx`.
+Click *Program* and wait for programming to complete.
+
+You should now see a window with one tab per ILA.
+Each tab has a waveform viewer at the top and two setting windows at the bottom.
+The left bottom window has two tabs.
+In the *Settings* tab, useful settings are mainly the number of windows and the position of the trigger in the window.
+In the *Status* tab, the ILA trigger can be armed, disarmed, and activated immediately.
+The right bottom window also has two tabs; the first defines the trigger conditions and the second the capture conditions (both based on probe signals of the ILA).
+Configure each ILA according to your needs and hit the 'play' button in the *Status* tab to arm its trigger.
+
+Last but not least, you need to run a test in which the FPGA does not get reprogrammed (otherwise the ILA or at least its configuration is lost).
+To this end, modify `sw/host/opentitanlib/src/test_utils/init.rs` to not load a bitstream:
```

Comment:
Technically this only applies to tests that have a custom host-side opentitanlib-linked test harness. For tests that just use opentitantool to drive them, you would need to comment out [this](https://cs.opensource.google/opentitan/opentitan/+/master:rules/opentitan_test.bzl;drc=bbf0ea10cdfb6681c094894ae00d0fe471c3edfc;l=200) line instead.

Reply:
Fixed, thanks!